src/thermostate/plotting.py

```diff
@@ -58,10 +57,15 @@ def plot(self, x_axis, y_axis):  # pragma: no cover
         """Hold the place of a plot function that a child class must establish."""
         pass
 
-    def add_state(self, state, key=None):
+    def add_state(self, state, key=None, label = None):
         """Add a state to the self.states dictionary and plot it."""
         if key is None:
             key = repr(state)
+        if label:
+            if isinstance(label, int) or isinstance(label, str):
+                state._label = label
+            else:
+                raise TypeError("Label must be an int or str")
```

Comment:
Python style is usually (not always) "it's easier to ask forgivness than permission" (sometimes abbreviated EAFP). This means in Python you'll more commonly see an action attempted, and any exceptions caught and handled. Something like this:

```python
try:
    state._label = str(label)
except Exception:
    raise TypeError(f"The given label '{label}' could not be converted to a string") from None
```

Reply:
And actually, since I'm recommending casting to a string for the label, this `try...except` can go into the setter for the `state.label` attribute (see below).